src/wp-includes/link-template.php

```diff
@@ -1739,6 +1739,25 @@ function edit_bookmark_link( $link = '', $before = '', $after = '', $bookmark =
  * @return string URL to edit user page or empty string.
  */
 function get_edit_user_link( $user_id = null ) {
+
+	/**
+	 * Short-circuits the retrieval of the user edit link.
+	 *
+	 * Returning a non-empty string from this filter will effectively short-circuit
+	 * the function, returning the passed value instead of continuing with the
+	 * standard link generation process.
+	 *
+	 * @since 6.8.0
+	 *
+	 * @param string|null $link    The edit link. Default is an empty string.
+	 * @param int        $user_id  User ID.
+	 */
+	$link = apply_filters( 'pre_get_edit_user_link', '', $user_id );
+
+	if ( ! empty( $link ) ) {
+		return $link;
+	}
```

Comment:
```suggestion
	 * Returning a string from this filter will effectively short-circuit the function,
	 * returning the passed value instead of continuing with the standard link
	 * generation process.
	 *
	 * @since 6.8.0
	 *
	 * @param string|null $link    The edit link. Default is an empty string.
	 * @param int         $user_id  User ID.
	 */
	$link = apply_filters( 'pre_get_edit_user_link', null, $user_id );

	if ( null !== $link ) {
		return $link;
	}
```


Imho this would be a little more strict and it would also allow to return an empty link, if anyone fancies doing so. Besides that I like the solution